osu.Game/Screens/SelectV2/BeatmapSetPanel.cs

```diff
@@ -69,11 +289,159 @@ protected override void PrepareForUse()
             Debug.Assert(Item != null);
             Debug.Assert(Item.IsGroupSelectionTarget);
 
-            var beatmapSetInfo = (BeatmapSetInfo)Item.Model;
+            var beatmapSet = (BeatmapSetInfo)Item.Model;
+
+            // Choice of background image matches BSS implementation (always uses the lowest `beatmap_id` from the set).
+            background.Beatmap = beatmaps.GetWorkingBeatmap(beatmapSet.Beatmaps.MinBy(b => b.OnlineID));
+
+            titleText.Text = new RomanisableString(beatmapSet.Metadata.TitleUnicode, beatmapSet.Metadata.Title);
+            artistText.Text = new RomanisableString(beatmapSet.Metadata.ArtistUnicode, beatmapSet.Metadata.Artist);
+            updateButton.BeatmapSet = beatmapSet;
+            statusPill.Status = beatmapSet.Status;
+
+            singleDiffStarsToken?.Cancel();
+            singleDiffStarsToken = new CancellationTokenSource();
+            singleDiffStarsBindable = null;
+
+            if (beatmapSet.Beatmaps.Count == 1)
+            {
+                chevronIcon.Hide();
+                difficultiesDisplay.Hide();
+
+                var singleBeatmap = beatmapSet.Beatmaps.Single();
+
+                singleDiffIcon.Icon = singleBeatmap.Ruleset.CreateInstance().CreateIcon();
+                singleDiffIcon.Show();
 
-            text.Text = $"{beatmapSetInfo.Metadata}";
+                singleDiffRank.Beatmap = singleBeatmap;
```

Comment:
Set to a beatmap but never cleaned up. This means that if a pooled panel goes unused, or worse, it comes back as a non-single-difficulty, it will keep a realm subscription alive indefiniteliy.